### PR TITLE
[Bugfix] ActiveRecord::StatementInvalid: OCIError: ORA-04043: object SP_INVOICES_FRIENDLY_ID does not exist

### PR DESCRIPTION
**Pull request by Martouta**
fixes https://github.com/3scale/porta/issues/75

### CODE REVIEW

File: lib/system/database.rb [hunk 1]
```diff
@@ -133,7 +133,7 @@ def drop
              EXECUTE IMMEDIATE 'DROP TRIGGER #{name}';
           EXCEPTION
             WHEN OTHERS THEN
-              IF SQLCODE != -4080 THEN
+              IF SQLCODE != -4043 THEN
```
Comment (hallelujah) [inline lib/system/database.rb]: 4080 should work for trigger

File: lib/system/database.rb [hunk 2]
```diff
@@ -252,7 +252,7 @@ def drop
              EXECUTE IMMEDIATE 'DROP PROCEDURE #{name}';
           EXCEPTION
             WHEN OTHERS THEN
-              IF SQLCODE != -4080 THEN
+              IF SQLCODE != -4043 THEN
```
Comment (hallelujah) [inline lib/system/database.rb]: Yes it is that one easy :)

➡️ `ORA-04043: object SP_INVOICES_FRIENDLY_ID does not exist` 

➡️  **ORA-04043**
➡️ 4043

**Comment by Martouta** [comment]
Hery found the 2nd issue 😍 
image

and it works:
image

File: lib/system/database.rb [hunk 3]
```diff
@@ -290,6 +290,7 @@ def create
       def create
         super
         connection.execute "GRANT create trigger TO #{username}"
+        connection.execute "GRANT create procedure TO #{username}"
```
Comment (guicassolato) [inline lib/system/database.rb]: Oops. Sorry, guys. My bad. Didn't see that.

**Comment by hallelujah** [comment]
OK too bad failing on memory allocation
At least it passes the DB setup
So good to be merged IMHO

**Pull request merged**